src/extras/core/CurvePath.js

```diff
@@ -66,7 +66,13 @@ THREE.CurvePath.prototype = Object.assign( Object.create( THREE.Curve.prototype
 				var diff = curveLengths[ i ] - d;
 				var curve = this.curves[ i ];
 
-				var u = 1 - diff / curve.getLength();
+				var segmentLength = curve.getLength();
+				var u = 0;
+				if ( segmentLength !== 0 ) {
+
+				  u = 1 - diff / curve.getLength();
```

Comment:
You are calling `curve.getLength()` twice.

    var u = ( segmentLength == 0 ) ? 0 : 1 - diff / segmentLength;

And what was the reason you added this block? To handle a degenerate case?